WordPress/Classes/ViewRelated/QR Login/QRLoginVerifyAuthorizationViewController.swift

```diff
@@ -1,10 +1,115 @@
 
+
+    @IBOutlet weak var confirmButton: UIButton!
+    @IBOutlet weak var cancelButton: UIButton!
+
+    var coordinator: QRLoginVerifyCoordinator?
+}
+
+// MARK: - View Methods
+extension QRLoginVerifyAuthorizationViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        // Do any additional setup after loading the view.
+        coordinator?.start()
+
+        applyStyles()
+    }
+
+    @IBAction func didTapConfirm(_ sender: Any) {
+        coordinator?.confirm()
+    }
+
+    @IBAction func didTapCancel(_ sender: Any) {
+        coordinator?.cancel()
+    }
+}
+
+// MARK: - QRLoginVerifyView
+extension QRLoginVerifyAuthorizationViewController: QRLoginVerifyView {
+    func render() {
+        imageView.image = UIImage(named: Strings.imageName)
+        titleLabel.text = Strings.title
+        subTitleLabel.text = Strings.subtitle
+        confirmButton.setTitle(Strings.confirmButton, for: .normal)
+        cancelButton.setTitle(Strings.cancelButton, for: .normal)
+
+        stackView.isHidden = false
+
+        loadingIndicator.isHidden = true
+        loadingIndicator.stopAnimating()
+    }
+
+    func renderCompletion() {
+        imageView.image = UIImage(named: Strings.completed.imageName)
+        titleLabel.text = Strings.completed.title
+        subTitleLabel.text = Strings.completed.subtitle
+        subTitleLabel.textColor = .secondaryLabel
+        confirmButton.setTitle(Strings.completed.confirmButton, for: .normal)
+        cancelButton.isHidden = true
+
+        stackView.layer.opacity = 1
+        loadingIndicator.isHidden = true
+        loadingIndicator.stopAnimating()
+
+        UINotificationFeedbackGenerator().notificationOccurred(.success)
+
+        ConfettiView.cleanupAndAnimate(on: view, frame: navigationController?.view.frame ?? view.frame) { confettiView in
+            // removing this instance when the animation completes, will prevent
+            // the animation to suddenly stop if users navigate away early
+            confettiView.removeFromSuperview()
+        }
+    }
+
+    func showLoading() {
+        stackView.isHidden = true
+        loadingIndicator.startAnimating()
+        loadingIndicator.isHidden = false
+    }
+
+    func showAuthenticating() {
+        stackView.layer.opacity = 0.5
+
+        loadingIndicator.startAnimating()
+        loadingIndicator.isHidden = false
+    }
+
+    func showNoConnectionError() {
+        // TODO: Add error handling
+        print("no connection error")
+    }
+}
+
+// MARK: - Private: View Helpers
+extension QRLoginVerifyAuthorizationViewController {
+    private func applyStyles() {
+        titleLabel.font = WPStyleGuide.serifFontForTextStyle(.title1, fontWeight: .semibold)
+        titleLabel.textColor = .text
+
+        subTitleLabel.font = .preferredFont(forTextStyle: .headline)
+        subTitleLabel.textColor = .systemRed
+    }
+
+    private struct Strings {
```

Comment:
[nit] I think enums make better containers for constants over structs since they're pure namespaces and can't be instantiated
```suggestion
    private enum Strings {
```

Reply:
Done in: 9464a3ce8eddb4dbd9fa52c681151c1739a79bcf